core/src/main/java/nl/nn/adapterframework/filesystem/FileSystemActor.java

```diff
@@ -449,7 +450,67 @@ protected void finalize() throws Throwable {
 
 		return input;
 	}
-	
+
+	/**
+	 * Helper method to process move and copy actions.
+	 */
+	private String processAction(Message input, ParameterValueList pvl, String destinationFolder, String action) throws FileSystemException, IOException {
+		if(StringUtils.isNotEmpty(getWildCard()) || StringUtils.isNotEmpty(getExcludeWildCard())) { 
+			String folder = arrangeFolder(determineInputFoldername(input, pvl));
+			XmlBuilder dirXml = new XmlBuilder(action+"FilesList");
+			try(Stream<F> stream = FileSystemUtils.getFilteredList(fileSystem, folder, getWildCard(), getExcludeWildCard())) {
+				Iterator<F> it = stream.iterator();
+				while(it.hasNext()) {
+					F file = it.next();
+					F processedFile = null;
+					switch(action) {
+						case ACTION_MOVE:
+							processedFile = FileSystemUtils.moveFile(fileSystem, file, destinationFolder, isOverwrite(), getNumberOfBackups(), isCreateFolder());
+							break;
+						case ACTION_COPY:
+							processedFile = FileSystemUtils.copyFile(fileSystem, file, destinationFolder, isOverwrite(), getNumberOfBackups(), isCreateFolder());
+							break;
+						default:
+							break;
+					}
+					if(processedFile != null) {
+						dirXml.addSubElement(getFileAsXmlBuilder(processedFile, "file"));
+					}
```

Comment:
My idea was not to have this switch(action) here, but to have action as a Function() parameter